### ISSUE TITLE
Add OpenM++ to remote desktop

### ISSUE THREAD

**Issue by goatsweater** [StatCan/aaw-kubeflow-containers#308]
SAMD and HAD both have a need for using OpenM++ for modelling work.

OpenM++ seems to have a fairly straight forward installation system (it just runs out of a directory), but requires other packages used to compile the generated C++ code.

- [x] Install OpenM++
- [x] Install dependencies
- [x] Run test model(s) to verify functionality

**Comment by blairdrummond** [comment]
@brendangadd @chuckbelisle this is for  oncosim

**Comment by goatsweater** [comment]
Running into the same build time errors with `vsce` for the remote-desktop image as #305. Applying fix from https://github.com/StatCan/aaw-kubeflow-containers/blob/2f689f24dcc06f43ce6a52a429e903d8922bc62a/docker-bits/6_remote-desktop.Dockerfile#L236 resolves the issue.

**Comment by goatsweater** [comment]
OpenM++ is shipped as a self contained `.tar.gz` file. This unpacks to an entire directory containing binaries and configuration files. To avoid polluting the file system I'm installing the entire package into `/opt/openmpp`.

**Comment by goatsweater** [comment]
Everything seems to be functioning. Added icon to the desktop for the UI and was able to run the samples from the docs.

Awaiting a model from the client that can be used as a test before submitting the pull request.

**Comment by goatsweater** [comment]
Got access to the OncoSim model for testing. Using that to verify everything is functional. So far only found `sqlite3` package to be missing.

**Comment by goatsweater** [comment]
The OncoSim model requires adjusting the `ulimit` or else it will error out with a Segmentation Fault. The `ulimit` default of 8192 is too low, and based on feedback from the project team needs to be increased to 65536.

Limit was increased through the command `ulimit -S -s 65536`. After that the segmentation fault was no longer experienced.

**Comment by blairdrummond** [comment]
@Jose-Matsuda do we just need to re-run the CI now?

**Comment by Jose-Matsuda** [comment]
@blairdrummond oh it was already pushed yesterday when Rohan commented out the container scan bit of the CI. He reverted it this morning to have the container scan back in business and remote-desktop did succeed so this should be available now. (the one that failed is the pytorch image because it was waiting on a pillow update to the conda-forge feedstock

**Comment by blairdrummond** [comment]
Is it now the default image?

**Comment by Jose-Matsuda** [comment]
Nope, here is the sha though `6b3b5375` (this is the sha from yesterday when we commented out the container scan, so this will work for the pytorch image as well)

**Comment by blairdrummond** [comment]
@goatsweater Try Custom image

```
k8scc01covidacr.azurecr.io/remote-desktop:6b3b5375
```

**Comment by rohank07** [comment]
@chuckbelisle can you update the shas in the jupyter-api config-map?

**Comment by goatsweater** [comment]
I created a machine based on the custom image and ran through a complete build/run cycle of the model. Everything worked well from my testing.

Question: is anyone aware of a way to inject the `ulimit -S -s 65536` command into only the remote desktop image? When I tried to add it to `.bashrc` it ended up applying to all images universally so I backed out the change. Maybe we want people to have to set it themselves rather than being applied globally?
